kuryr_kubernetes/controller/drivers/namespace_security_groups.py

```diff
@@ -105,18 +106,50 @@ def create_namespace_sg(self, namespace, project_id, crd_spec):
                 {
                     "security_group": {
                         "name": sg_name,
-                        "project_id": project_id
+                        "project_id": project_id,
+                        "description": "Kuryr Namespace SG"
                     }
                 }).get('security_group')
             utils.tag_neutron_resources('security-groups', [sg['id']])
+            # NOTE(ltomasbo): Neutron populates every new SG with two rules
+            #                 allowing egress on IPv4 and IPv6. As namespace
+            #                 isolation does not handle egress, we remove them
+            #                 and leave egress for the default pods SG, or
+            #                 open to modifications per namespace.
+            for sgr in sg['security_group_rules']:
+                neutron.delete_security_group_rule(sgr['id'])
+
             neutron.create_security_group_rule(
                 {
                     "security_group_rule": {
                         "direction": "ingress",
                         "remote_ip_prefix": crd_spec['subnetCIDR'],
-                        "security_group_id": sg['id']
+                        "security_group_id": sg['id'],
+                        "description": "Kuryr Namespace SG rule"
                     }
                 })
+            # Adding rules to the default SG group if global namespace
+            if namespace in cfg.CONF.namespace_sg.global_namespaces:
+                neutron.create_security_group_rule(
+                    {
+                        "security_group_rule": {
+                            "direction": "ingress",
+                            "remote_ip_prefix": crd_spec['subnetCIDR'],
+                            "security_group_id": (
+                                cfg.CONF.namespace_sg.sg_allow_from_default)
+                        }
+                    })
+                neutron.create_security_group_rule(
```

Comment:
The first rule you add doesn't mention any protocol, so most likely will open it for any. Why is the udp one needed?

Reply:
right... I think while testing it was only creating the tcp one... so I added the udp one... but now it seems it is only adding the rule without protocol... so. I'll double check